### ISSUE TITLE
Rails 6 support

### ISSUE THREAD

**Issue by andreimoment** [RolifyCommunity/rolify#510]
With Rails 6 around the corner I am wondering if any work is being done to verify support?

Any reason for concerns with using rolify + Rails 6?

**Comment by stevehill1981** [comment]
Just tried this myself this morning against Rails 6 master.

The default migrations don't work:

```
$ bundle exec rails db:migrate
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:

  class RolifyCreateRoles < ActiveRecord::Migration[4.2]
```

It seems to work fine when updated to include the Rails version identifier. I haven't finished integrating the system with my code yet, so I may run into other difficulties.

**Comment by andreimoment** [comment]
@stevehill1981 thank you! Please consider updating this thread with any issues, will make it easier to update the code.

**Comment by stevehill1981** [comment]
This is literally the only problem I've found so far.

I've only got a very basic skeleton application, but all of the other rolify functionality seems to work as expected.

**Comment by slbug** [comment]
Using Rails 6.0.0.rc1 I'm receiving deprecation warning if rolify initializer present.

```
DEPRECATION WARNING: Initialization autoloaded the constants ApplicationRecord and Role.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.

Reloading does not reboot the application, and therefore code executed during
initialization does not run again. So, if you reload ApplicationRecord, for example,
the expected changes won't be reflected in that stale Class object.

These autoloaded constants have been unloaded.

Please, check the "Autoloading and Reloading Constants" guide for solutions.
```

**Comment by stevehill1981** [comment]
Is this in Zeitwerk mode, or classic? Zeitwerk is new in Rails 6, and is enabled as standard, so that's probably what you've got.

I'm not sure how to fix this deprecation, but I've not really looked into it yet.

**Comment by slbug** [comment]
Zeitwerk

**Comment by epipheus** [comment]
Any eta on rails 6 compatibility?

**Comment by rajdeepbhatia** [comment]
Any update on Rails 6 compatibility guys?

**Comment by mepatterson** [comment]
> Is this in Zeitwerk mode, or classic? Zeitwerk is new in Rails 6, and is enabled as standard, so that's probably what you've got.
> 
> I'm not sure how to fix this deprecation, but I've not really looked into it yet.

FWIW, I think you may be able to eliminate the deprecation warning by wrapping everything in your initializer with:
```
Rails.configuration.to_prepare do
  ... all the stuff
end
```
CAVEAT: I haven't tried this yet, but had a similar problem with a similar bit of internal code in our app and that made zeitwerk happy. I'm not claiming it's the right solution, though...

**Comment by stevehill1981** [comment]
This problem doesn't seem to exist for me with a newly generated Rails 6.0.2.2 app using Rolify's `master` branch. I'll be pushing an example of that shortly.

I've made a basic fork, which I'll also be pushing shortly, that fixes the migration generator. So far, that's the only issue I've discovered, but admittedly my example app is extremely simple so there are undoubtedly scenarios I've not covered.

**Comment by stevehill1981** [comment]
I've pushed a test app here: https://github.com/stevehill1981/rolify-rails6-test

I've also forked Rolify so I can work on it privately: https://github.com/stevehill1981/rolify

That fork includes the fix for the Rails migration generator, which is the only issue I found.

**Comment by stevehill1981** [comment]
I've created a fork which resolves the migration generator issue - that's over at stevehill1981/rolify.

I'll PR that here shortly, but the fix is essentially the same as my last PR which never ended up being merged.

**Comment by kkurian** [comment]
> Using Rails 6.0.0.rc1 I'm receiving deprecation warning if rolify initializer present with `Rolify.configure('Role')...`. But no warnings if configure called without params
> 

To resolve the above [issue](https://github.com/RolifyCommunity/rolify/issues/510#issuecomment-510847280) reported by @slbug see https://github.com/rails/rails/issues/36363

TL;DR Do something like this: Move config/initializers/rolify.rb to app/lib/rolify.rb (or anywhere in the `ActiveSupport::Dependencies.autoload_paths`) then create a new config/initializers/rolify.rb containing just `require 'rolify'`
